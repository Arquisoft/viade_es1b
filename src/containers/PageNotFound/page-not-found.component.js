/* eslint-disable constructor-super */
import React from "react"
import { useTranslation } from "react-i18next"
import {
  PageNotFoundWrapper,
  PageNotFoundContent,
} from "./page-not-found.style"

/**
 * A React component page that is displayed when there's no valid route. Users can click the button
 * to get back to the home/welcome page.
 */
const PageNotFound = () => {
  const { t } = useTranslation()
  return (
    <PageNotFoundWrapper data-testid="page-not-found-container">
      <PageNotFoundContent>
        <img src="/img/404.svg" alt="404" />
        <h3>{t("notFound.title")}</h3>
        <p>{t("notFound.content")}</p>
        <div>
          <a
            href="/"
            id="linkToHomepage"
            className="ids-link"
            data-testid="page-not-found-homepage"
          >
            {t("notFound.redirectButton")}
          </a>
        </div>
      </PageNotFoundContent>
    </PageNotFoundWrapper>
  )
}

export default PageNotFound
