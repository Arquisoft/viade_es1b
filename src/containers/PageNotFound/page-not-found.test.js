import React from 'react';
import { render, cleanup, queryByTestId, queryByAltText, fireEvent } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import PageNotFound from './page-not-found.component';

let wrapper;
beforeEach(() => {
  const { container, debug } = render(
    <Router>
      <PageNotFound />
    </Router>
  );
  wrapper = container;
    debug();
});

describe('Page Not Found', () => {
  afterAll(cleanup);

  test('App renders without crashing', () => {
    expect(wrapper).toBeTruthy();
  });

  
  test('includes link to homepage', () => {

    expect(queryByTestId(wrapper, "page-not-found-homepage")).not.toBeNull();
    
  });


});
