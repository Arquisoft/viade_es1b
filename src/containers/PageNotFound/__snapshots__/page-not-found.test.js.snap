// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Not Found Page Render renders with default props 1`] = `ShallowWrapper {}`;
